Packs/Vectra/Integrations/VectraEventCollector/VectraEventCollector_test.py

```diff
@@ -0,0 +1,404 @@
+"""
+Unit tests for Vectra Event Collector
+"""
+
+import pytest
+from pytest_mock import MockerFixture
+from VectraEventCollector import (
+    VectraClient,
+    is_eod,
+    test_module,
+    DETECTION_FIRST_TIMESTAMP_QUERY_START_FORMAT,
+    AUDIT_START_TIMESTAMP_FORMAT,
+    DETECTION_FIRST_TIMESTAMP_FORMAT,
+    DETECTION_NEXT_RUN_KEY,
+    AUDIT_NEXT_RUN_KEY,
+    get_detections_cmd,
+    get_audits_cmd,
+    get_events,
+    fetch_events,
+)
+from typing import Dict, Any
+import json
+from datetime import datetime
+from pathlib import Path
+from CommonServerPython import *
+from hypothesis import given, strategies as st
+from freezegun import freeze_time
+
+""" Constants """
+BASE_URL = "mock://dev.vectra.ai"
+PASSWORD = "9455w0rd"
+client = VectraClient(url=BASE_URL, api_key=PASSWORD)
+
+
+def load_json(path: Path):
+    with open(path, mode="r", encoding="utf-8") as f:
+        return json.load(f)
+
+
+AUDITS: Dict[str, Any] = load_json(Path("./test_data/audits.json"))
+DETECTIONS: Dict[str, Any] = load_json(Path("./test_data/search_detections.json"))
+endpoints = load_json(Path("./test_data/endpoints.json"))
+no_access_endpoints = load_json(Path("./test_data/endpoints_no_detection_audits.json"))
+
+""" VectraClient Tests """
+
+
+@pytest.mark.parametrize(
+    "endpoints,expected",
+    [
+        ({"detections", f"{BASE_URL}/detections", "audits", f"{BASE_URL}/audits"}, True),
+        ({"detections", f"{BASE_URL}/detections"}, False),
+        ({"ep1", f"{BASE_URL}/ep1"}, False),
+        ({}, False),
+    ],
+)
+def test_auth(mocker: MockerFixture, endpoints: Dict[str, str], expected: bool):
+
+    """
+    Given:
+        - A Vectra client.
+    When:
+        - Case A: The returned endpoints from the API root are the required ones.
+        - Case B: The returned endpoints from the API root are missing 'audits'.
+        - Case C: The returned endpoints from the API root are missing 'audits' and 'detections'.
+        - Case D: The returned endpoints from the API root is empty.
+    Then:
+        - Case A: The authentication should succeed
+        - Case B: The authentication should fail
+        - Case C: The authentication should fail
+        - Case D: The authentication should fail
+    """
+
+    mocker.patch.object(client, "get_endpoints", return_value=endpoints)
+    endpoints = client.get_endpoints()
```

Comment:
You are not really testing anything here, it is not really needed to test client function especially when all they do is perform API requests.